### DISCUSSION TITLE
Graph RAG: 知识图谱结合 LLM 的检索增强 - siwei.io

### DISCUSSION THREAD

**Discussion by giscus** [siwei-io/siwei-io.github.io#29] [Comments]
# Graph RAG: 知识图谱结合 LLM 的检索增强 - siwei.io

Graph RAG， LLM 开发的图谱驱动新技术栈与上下文学习方式

https://siwei.io/graph-rag/

**Comment by SpikeWong** [comment]
最近在做RAG, 感谢分享

**Reply by wey-gu** [reply]
感谢，欢迎来交流哈。

**Comment by SpikeWong** [comment]
> 最后，我做了一个 streamlit 的 demo来比较 Graph RAG 与 Vector RAG，从中我们可以看到 Graph RAG 并没有取代 Embedding、向量搜索的方法，而是增强了/补充了它的不足。

这里我在文中没有找到原因，因为目前也设计 RAG 的方案，所以想了解下您的一些看法。
在我的理解里，目前 Graph RAG 能够处理主体找属性和属性找主体的场景，但是无法或者很难做文本内容的相似度检索，所以基于相似度检索场景和图检索场景走不同的召回通路。想问下我理解的对吗？

**Reply by wey-gu** [reply]
我在其他的地方有分享过：
- Graph 的数据对数据的分割和获取是更细粒度的，这个和 vector 的相对粗粒度 chunking 是一个补充
- KG 中保有数据中的跨篇幅、全局连接，能够被捞取出来，满足一些 rag 任务
- KG 中保有一些知识是能 mitigate embedding 方法带来的幻觉的

ref: https://bit.ly/graph-rag-workshop

**Comment by chenlixuan** [comment]
从文中我理解的graph rag，是通过初步检索到的点，通过几跳查询去扩展检索到的信息，也就是丰富了检索出来的上下文信息，因为纯向量检索的话，可能获取的上下文信息不够全面。如果对于多跳查询的场景来说，有没有可能从另一个角度入手，也就是对问题进行重写，让问题更加明确，这样通过向量检索出来的上下文信息已经足够丰富

**Comment by Julian-Huang-1** [comment]
大佬，你在亚马逊云科技大会上分享了Graph RAG，视频地址：https://www.bilibili.com/video/BV1dH4y1z7Tk/?spm_id_from=333.337.search-card.all.click&vd_source=2b529b02e53568bba44d93239340e995，
在视频时间43:00处分享了一篇论文关于reasoning over graph，我没找到这篇论文，可以分享下论文的标题吗？或者论文arxiv链接也可以。谢谢大佬！！！

**Reply by wey-gu** [reply]
https://arxiv.org/abs/2310.01061 :)

也敬请期待我们的 CoE，现在还没公开开源实现，但是我在 PyCon China 上做了 demo。

**Comment by LovingPastry** [comment]
老师您好！我是一名正在学习GraphRAG的大学生，看到您的博客与公布在b站上的视频学到了很多知识，十分感谢！针对视频演示以及本项目中提到的，使用LLM生成text2cypher或nl2cypher函数，本质上是将用户的问句和生成cypher语句的指令一起作为prompt输送给大模型，那么请问会不会遇到大模型生成的关键字段不准确而导致无法查询到答案呢？比如大模型返回的cypher查询语句中的label是"登录"，但我的数据库中的label是“登陆”，因为“录”字不一样而导致的无法查询。

感谢老师分享的这么好的成果与这么优质的课程！如果老师能在百忙之中回答我的问题，我将感激不尽！

**Comment by wey-gu** [comment]
https://www.siwei.io/demo-dumps/graph-rag/GraphRAG.html

**Comment by codeinging** [comment]
老师您好，感谢您的分享，有个问题请教一下，希望能够得到您的解答，您在文章中提到结合text2cypher的Graph RAG，我在 https://docs.llamaindex.ai/en/stable/examples/query_engine/knowledge_graph_rag_query_engine/ 中查到llamaindex已经支持了，但我现在的项目是基于Langchain的，请问Langchain有没有类似的支持呢？谢谢！

**Comment by mwx66** [comment]
老师，您好，请问llamaindex中Graph RAG，可以基于本地化部署的LLM做吗？

**Comment by wangqiuxiao** [comment]
老师，你好，KnowledgeGraphRAGQueryEngine这个目前支持了吗。
